frontend/src/components/pages/RegistrantExperience/SessionSelection/index.tsx

```diff
@@ -0,0 +1,177 @@
+import React from "react";
+import {
+  Button,
+  Box,
+  Image,
+  Text,
+  Show,
+  Flex,
+  AspectRatio,
+} from "@chakra-ui/react";
+import { CampResponse } from "../../../../types/CampsTypes";
+import CampDetailsSummary from "./CampDetailsSummary";
+import CampSessionBoard from "./CampSessionBoard";
+import { SessionSelectionState } from "./SessionSelectionTypes";
+
+type SesssionSelectionProps = {
+  camp: CampResponse;
+  selectedSessions: Set<string>;
+  setSelectedSessions: (newSelections: Set<string>) => void;
+  onFormSubmission: () => void;
+};
+
+const SessionSelection = ({
+  camp,
+  selectedSessions,
+  setSelectedSessions,
+  onFormSubmission,
+}: SesssionSelectionProps): React.ReactElement => {
+  // Set the selection mode - all selections should be waitlisted sessions
+  // or all selections should be available sessions
+  let selectionState = SessionSelectionState.None;
```

Comment:
Do we need the None state? If I understand correctly, the following code just sets the state to be the state of the first session. Could we not wrap this in a useEffect and initialize the selectionState to the state of the first session by doing: 

```
let selectionState = camp.campSessions[0].campers.length === camp.campSessions[0].capacity ? SessionSelectionState.SelectingWaitlistSessions : SessionSelectionState.SelectingAvailableSessions ; 
```

do we want the selectionState to be recalculated on every render?

Reply:
I changed this to a state variable so it wouldn't be calculated on every render, only when (1) you have no sessions and add a session or (2) when you remove sessions so that you have none selected. 

It can't be completely unchangeable because of (2)